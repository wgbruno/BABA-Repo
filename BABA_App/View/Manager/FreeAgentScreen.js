import React, { useEffect, useState } from "react";
import { NavigationContainer } from 'react-native';
import { View, Text, Image, StyleSheet, useWindowDimensions, ScrollView, Alert, FlatList, TouchableOpacity } from 'react-native';
import { UserState } from "realm";
import Logo from '../../assets/images/Logo_1.png'
import { getTeamPlayers } from "../../DAOs/PlayerDao";


export default function FreeAgentScreen({navigation}){
    const {height} = useWindowDimensions();
    const [players, setPlayers] = useState(getTeamPlayers("Free Agent"));
    const [empty, setEmpty] = useState(true);
    const [userName, setUser] = useState("");

    useEffect(() => {
        console.log(navigation.getParam('paramUserName', "N/A"));
        setUser(navigation.getParam('paramUserName', "N/A"))
        if(players != undefined){
            setEmpty(false);
        }
    })

    const noFreeAgents = () => {
        return(
            <View style={{justifyContent: 'center', alignItems: 'center', flex: 1}}>
                <Text style={{fontSize: 25, textAlign: 'center', color: 'darkblue'}}>
                    There are no available free agents.
                </Text>
            </View>
        );
    }
    return (
        <View style={styles.root}>
        <Image
            source={Logo}
            style={[styles.logo, {height: height * 20}]}
            resizeMode="contain"
        />
        <Text style={{marginTop: 8, fontWeight: 'bold', color: 'darkblue', fontSize: 20}}>Free Agents</Text>
        {empty ? noFreeAgents():
            <FlatList
                data={players}
                keyExtractor={(item, index) => index.toString()}
                renderItem={({item, index}) => 
                    <View style={{flexDirection: 'row', justifyContent: 'space-between'}}>
                        <TouchableOpacity
                            onPress={() => {navigation.navigate('SendTeamRequest', {
                                paramPlayerID: item.playerID,
                                paramManagerID: userName,    
                            });}}>
                            <Text style={styles.list}>{item.firstName} {item.lastName}   Team: {item.teamName} ID: {item.playerID}</Text>
                            <Text style={styles.list}>Age: {item.age}      Height: {item.height}      Number: {item.number}</Text>
                        </TouchableOpacity>
                    </View>
                } />
            }
        </View>
    );
};

const styles = StyleSheet.create({
    list: {
        marginTop: 8, 
        fontWeight: 'bold', 
        color: 'red', 
        fontSize: 20
    },
    root: {
      alignItems: 'center',
      padding: 10,
      flex: 1
    },
    logo: {
      width: '3000%',
      maxWidth: 200,
      maxHeight: 150,
    },
  });