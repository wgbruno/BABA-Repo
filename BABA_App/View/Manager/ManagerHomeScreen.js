import React, { useEffect, useState } from "react";
import { NavigationContainer } from 'react-native';
import CustomButton from "../../src/components/CustomButton";
import { View, Text, Image, StyleSheet, useWindowDimensions, ScrollView, Alert } from 'react-native';
import Logo from '../../assets/images/Logo_1.png'
import { Account } from "../../Objects/AccountCont";
import { Player } from "../../Objects/PlayerCont";
import { getNewIDDB } from "../../DAOs/PlayerDao";

export default function ManagerHomeScreen({navigation}){
  const {height} = useWindowDimensions();
  const [userName, setUser] = useState("");

  useEffect(() => {
    setUser(navigation.getParam('paramUserName', "test"));
  })

  const toTeam = () => {
    navigation.navigate("DisplayTeam", {paramTeamName: teamName});
  }
  
  const freeAgents = () => {
    navigation.navigate('FreeAgent', {paramUserName: userName});
  }
  
  
  return (
      <ScrollView showsVerticalScrollIndicator={false}>
        <View style={styles.root}>
          <Image
            source={Logo}
            style={[styles.logo, {height: height * 20}]}
            resizeMode="contain"
          />
          <CustomButton
            text={"View Team"}
            onPress={toTeam}
          />

          <CustomButton
            text={"Free Agents"}
            onPress={freeAgents}
          />
      </View>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
    root: {
      alignItems: 'center',
      padding: 20,
    },
    logo: {
      width: '3000%',
      maxWidth: 200,
      maxHeight: 150,
    },
  });