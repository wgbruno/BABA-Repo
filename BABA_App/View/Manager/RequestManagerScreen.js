import React, { useEffect, useState } from "react";
import { NavigationContainer } from 'react-native';
import CustomButton from "../../src/components/CustomButton";
import { View, Text, Image, StyleSheet, useWindowDimensions, ScrollView, Alert } from 'react-native';
import Logo from '../../assets/images/Logo_1.png'
import { Account } from "../../Objects/AccountCont";

export default function RequestManagerScreen({navigation}){
    const {height} = useWindowDimensions();
    const [userName, setUserName] = useState("");

    useEffect(() => {
        setUserName(navigation.getParam("paramUserName", "N/A"));
    })
    
    const toChangeAccount = () => {
        var tmp = new Account(userName);
        const account = tmp.getAccount();
        if(account.changeType()){
            Alert.alert("Failed!", "Failed to convert to manager account!");
        }else{
            Alert.alert("Success!", "Converted account to manager!");
            navigation.navigate("PlayerHome");
        }
    }
    

    return(
        
        <View style={styles.root}>
        <Image
            source={Logo}
            style={[styles.logo, {height: height * 20}]}
            resizeMode="contain"
        />
        <CustomButton
            text={"Convert " +userName+ " to manager"}
            onPress={toChangeAccount}
        />
        </View>
    )
};

const styles = StyleSheet.create({
    list: {
        marginTop: 8, 
        fontWeight: 'bold', 
        color: 'red', 
        fontSize: 20
    },
    root: {
      alignItems: 'center',
      padding: 10,
      flex: 1
    },
    logo: {
      width: '3000%',
      maxWidth: 200,
      maxHeight: 150,
    },
});