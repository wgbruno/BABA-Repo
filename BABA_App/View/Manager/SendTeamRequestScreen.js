import React, { useEffect, useState } from "react";
import { NavigationContainer } from 'react-native';
import { Player } from "../../Objects/PlayerCont";
import { Account } from "../../Objects/AccountCont";

export default function SendTeamRequestScreen({navigation}){
    const [playerID, setPlayerID] = useState("");
    const [managerID, setManagerID] = useState("");

    useEffect(() => {
        setPlayerID(navigation.getParam("paramPlayerID", 0));
        setManagerID(navigation.getParam("paramManagerID", "N/A"));
    })

    const toSendRequest = () => {
        
    }

};
