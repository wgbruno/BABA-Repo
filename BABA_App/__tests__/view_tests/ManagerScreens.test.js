import {expect, jest, test} from '@jest/globals';
import renderer from 'react-test-renderer';
import FreeAgentScreen from '../../View/Manager/FreeAgentScreen';
import ManagerHomeScreen from '../../View/Manager/ManagerHomeScreen';
import RequestManagerScreen from '../../View/Manager/RequestManagerScreen';
import SendTeamRequestScreen from '../../View/Manager/SendTeamRequestScreen';

test("Free Agent Screen", function () {
    const fa_tree = renderer.create(<FreeAgentScreen/>).toJSON();
    expect(fa_tree).toMatchSnapshot();
});
test("Manager Home Screen", function () {
    const tree = renderer.create(<ManagerHomeScreen/>).toJSON();
    expect(tree).toMatchSnapshot();
});
test(" Screen", function () {
    const tree = renderer.create(<RequestManagerScreen/>).toJSON();
    expect(tree).toMatchSnapshot();
});
test("Free Agent Screen", function () {
    const tree = renderer.create(<SendTeamRequestScreen/>).toJSON();
    expect(tree).toMatchSnapshot();
});