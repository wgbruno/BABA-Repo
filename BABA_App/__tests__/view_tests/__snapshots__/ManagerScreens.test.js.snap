// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Free Agent Screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "padding": 10,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../assets/images/Logo_1.png",
      }
    }
    style={
      Array [
        Object {
          "maxHeight": 150,
          "maxWidth": 200,
          "width": "3000%",
        },
        Object {
          "height": 26680,
        },
      ]
    }
  />
  <Text
    style={
      Object {
        "color": "darkblue",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginTop": 8,
      }
    }
  >
    Free Agents
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "darkblue",
          "fontSize": 25,
          "textAlign": "center",
        }
      }
    >
      There are no available free agents.
    </Text>
  </View>
</View>
`;
